sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1060,20 +1068,49 @@ private boolean isManagementToken(String lockToken) {
                     logger.info("Could not perform on session manger. Performing on management node.");
                     return performOnManagement;
                 });
-        }
-
-        final ServiceBusAsyncConsumer existingConsumer = consumer.get();
-        if (isManagementToken(lockToken) || existingConsumer == null) {
-            return performOnManagement;
         } else {
-            return existingConsumer.updateDisposition(lockToken, dispositionStatus, deadLetterReason,
-                deadLetterErrorDescription, propertiesToModify, transactionContext)
-                .then(Mono.fromRunnable(() -> {
-                    logger.info("{}: Update completed. Disposition: {}. Lock: {}.",
-                        entityPath, dispositionStatus, lockToken);
-                    renewalContainer.remove(lockToken);
-                }));
+            final ServiceBusAsyncConsumer existingConsumer = consumer.get();
+            if (isManagementToken(lockToken) || existingConsumer == null) {
+                updateDispositionOperation = performOnManagement;
+            } else {
+                updateDispositionOperation = existingConsumer.updateDisposition(lockToken, dispositionStatus,
+                    deadLetterReason, deadLetterErrorDescription, propertiesToModify, transactionContext)
+                    .then(Mono.fromRunnable(() -> {
+                        logger.info("{}: Update completed. Disposition: {}. Lock: {}.",
+                            entityPath, dispositionStatus, lockToken);
+                        renewalContainer.remove(lockToken);
+                    }));
+            }
         }
+        return updateDispositionOperation
+            .onErrorMap(throwable -> {
+                ServiceBusErrorSource errorSource;
+                if (throwable instanceof AmqpException) {
+                    switch (dispositionStatus) {
+                        case COMPLETED:
+                            errorSource = ServiceBusErrorSource.COMPLETE;
+                            break;
+                        case DEFERRED:
+                            errorSource = ServiceBusErrorSource.DEFER;
+                            break;
+                        case SUSPENDED:
+                            errorSource = ServiceBusErrorSource.DEAD_LETTER;
+                            break;
+                        case ABANDONED:
+                            errorSource = ServiceBusErrorSource.ABANDONED;
+                            break;
+                        default:
+                            throw logger.logExceptionAsError(new UnsupportedOperationException(String.format(
```

Comment:
Don't need to throw another exception here. If the `dispositionStatus` is unknown, we should just mark that as unknown and use that as the source.